controllers/front/success.php

```diff
@@ -47,19 +48,43 @@ public function postProcess()
 
             Tools::redirect($redirectLink);
         }
+        try {
+            $this->assertTransaction($orderId);
```

Comment:
should really check it's status as well like in successHosted

Reply:
assertTransaction already handles transaction logic. If it's canceled or the transaction failed for whatever reason it just returns bad response and throws an exception which in this case i catch the exception then redirect logic to failValidation controller where it updates status etc. Maybe if for whatever case the transaction fails but returns good response i double check the transaction status and throw exception myself ? ?